yt/utilities/lib/particle_kdtree_tools.pyx

```diff
@@ -134,26 +310,29 @@ cdef inline int cull_node(Node* node,
         else:
             tpos = 0
         ndist += tpos*tpos
-    return ndist > queue.heap_ptr[0]
+        k += axes.step
+
+    return (ndist > queue.heap[0] and queue.size == queue.max_elements)
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
 cdef inline int process_node_points(Node* node,
-                                    np.float64_t* pos,
-                                    np.float64_t[:, :] positions,
                                     BoundedPriorityQueue queue,
-                                    uint64_t skip_idx) nogil except -1:
-    cdef uint64_t i, idx
+                                    np.float64_t[:, ::1] positions,
+                                    np.float64_t* pos,
+                                    int skipidx,
+                                    axes_range * axes,
+                                    ) nogil except -1:
+    cdef uint64_t i, k
     cdef np.float64_t tpos, sq_dist
-    cdef int j
-    cdef np.float64_t* p_ptr
     for i in range(node.left_idx, node.left_idx + node.children):
-        p_ptr = &(positions[i, 0])
-        if i != skip_idx:
-            sq_dist = 0
-            for j in range(3):
-                tpos = p_ptr[j] - pos[j]
-                sq_dist += tpos*tpos
-            queue.add(sq_dist)
+        sq_dist = 0
+        k = axes.start
+        while k < axes.stop:
+            tpos = positions[i, k] - pos[k]
+            sq_dist += tpos*tpos
+            k += axes.step
```

Comment:
as referenced earlier this replaces range(3) to allow easy skipping of certain axes for projection plots which will be implemented soon